### PR TITLE
Chromatic polynomial

### PR DESCRIPTION
**Pull request by lucasmccabe**
This PR implements the chromatic polynomial, continuing development of [graph polynomial algorithms](https://networkx.org/documentation/stable/reference/algorithms/polynomials.html).

To follow the stack-based pattern of [#5265](https://github.com/networkx/networkx/pull/5265), I increment a `contraction_idx` graph attribute to track signs in the deletion-contraction expansion.

Code has been run through black and has been tested locally.

Thank you!

### CODE REVIEW

**Comment by lucasmccabe** [comment]
Hi @rossbar, do you have guidance wrt this PR? Thanks for your time!

File: networkx/algorithms/polynomials.py [hunk 1]
```diff
@@ -162,3 +162,130 @@ def tutte_polynomial(G):
...
+    import sympy
+
+    x = sympy.Symbol("x")
+    stack = deque()
+    stack.append(nx.MultiGraph(G, contraction_idx=0))
+
+    polynomial = 0
+    while stack:
+        G = stack.pop()
+        bridges = set(nx.bridges(G))
```
Comment (boothby) [inline networkx/algorithms/polynomials.py:278]: this appears to be unused

Comment (lucasmccabe) [reply networkx/algorithms/polynomials.py]: @boothby Yes! Thanks for catching this. I've corrected this in the recent commit [d3aac5b](https://github.com/networkx/networkx/pull/5675/commits/d3aac5bd9891ce5723de454bd3384a3be610e205)

**Review by boothby — APPROVED** [review]

**Comment by boothby** [comment]
Oh, I forgot to mention one thing.  Is it actually good to simplify the polynomial?  It doesn't look like `sympy` does a terribly good job factoring the polynomials -- it peels off an `x` from the complete graph whose polynomial is `prod(x-i for i in range(n)`, so I'm wondering if that's worth the effort.

**Comment by lucasmccabe** [comment]
@boothby That's a good point - it looks like the unsimplified polynomial is just as readable.

The simplification is omitted in [648f9ed](https://github.com/networkx/networkx/pull/5675/commits/648f9ed7d78605baa1e4d84e78f376932f82684a).

Thanks!

**Review by rossbar — APPROVED** [review]
Thanks @lucasmccabe , I took the liberty of adding the new function to the refguide, but other than that everything LGTM!

**Comment by lucasmccabe** [comment]
Thank you @rossbar and @boothby!

**Pull request merged**